### ISSUE TITLE
Input issues with multiple keyboards

### ISSUE THREAD

**Issue by louisrousseau** [Curtis-VL/OVRToolkit-Issues#143]
1. On a computer with multiple USB keyboards connected (in my case, two)
2. Start OVR Toolkit (on a fresh install)
3. Get the prompt to use a controller or press F8
4. Press F8 on one keyboard
5. Nothing happens
6. Press F8 on the other keyboard, works

I can work around the problem by disconnecting what it considers the "first" keyboard for now. (Long story short: I move a bunch of peripherals between my personal and work computers using a hub and an A-B switch; one keyboard + mouse are connected directly to the personal one at all times.)

**Comment by Curtis-VL** [comment]
Hi there!

This may be due to what the keyboard is defining itself as, OVR Toolkit is using DirectInput to detect devices, thus individual keyboards are individual devices. I'm not at my PC to check right now but if I recall it's supposed to bundle all 'keyboard' devices together. (Though it is kind of cool to have them separate in some instances to get more keys!)

If you could, please send your OVR Toolkit log file. This will be in AppData, LocalLow, Curtis English, OVR Toolkit. Look for the Player.log file. :)

**Comment by louisrousseau** [comment]
[Player.log](https://github.com/Curtis-VL/OVRToolkit-Issues/files/9448535/Player.log)

Ok... things are getting weirder, I got F8 to respond but not the arrow keys to navigate.
I'm in Steam VR home, used a controller and the in-VR Steam menu, library and started the overlay. Was able to get Enter and F8 to respond but the arrow keys don't seem to work; there's one of my monitor's (number 2 per Windows but the leftmost one) desktop being shown with a blue dot and the mouse following my head. To the left of that a (French-localized) "Create new window" tutorial:
![2022-08-29_01_OVRToolkit_screenshot](https://user-images.githubusercontent.com/892682/187320864-8579c90d-aadf-48de-93ee-e9a1d68c79d1.jpg)

**Comment by Curtis-VL** [comment]
Just a quick side note but OVR Toolkit will not work whilst the SteamVR Dashboard is open!

**Comment by louisrousseau** [comment]
Welp, looks like this is user error, works now, maybe head gaze is playing tricks on me with where it puts focus and the inputs being routed to either the overlay or the desktop. (SteamVR dashboard was closed at the moment.)

**Comment by Curtis-VL** [comment]
> Welp, looks like this is user error, works now, maybe head gaze is playing tricks on me with where it puts focus and the inputs being routed to either the overlay or the desktop. (SteamVR dashboard was closed at the moment.)

Woah woah woah, hold on. 😅

Just for my knowledge, did you manually rebind the keys so they would work on the other keyboard? Or was it just something like the SteamVR Dashboard being open?

When using OVR Toolkit with VR controllers it'll actually show a message on the wristwatch saying "Please close the SteamVR Dashboard to use OVR Toolkit", unfortunately there's no good place to display this message when using OVR Toolkit without VR controllers as it'd just be in the way. :)

**Comment by louisrousseau** [comment]
Meanwhile, I fired up a key tester and my main keyboard itself was acting up on _one_ key: F8 not getting presses 😐. Nice red herring that one. Reseated the buckling spring cap and it looks okay now. 

>Just for my knowledge, did you manually rebind the keys so they would work on the other keyboard? Or was it just something like the SteamVR Dashboard being open?

No rebinding. No SteamVR dashboard open in VR at that moment.

I reconnected the secondary keyboard and can interact with OVRToolkit with either keyboard without any apparent issue. They look like they're treated as a single bundle.

I think it's definitely head gaze, if I make sure to look outside of the virtual window I can interact with OVRToolkit correctly; if I don't and I try moving windows around as the tutorial explains I ended up inputting a bunch of adsswsws in this very text window which happened to be focused.

As a future improvement: May I suggest disabling inputs to the windows while they're being moved?

As an additional data point: I am using a WMR HP Reverb G2 so perhaps that might be playing a role? (It is set to the default "send inputs to the VR headset" rather than "send to desktop". If I remove the headset there's a blue bar at the top of the screen with text saying to press Win+Y to switch.)
